### PR TITLE
Don't fail when we can't overwrite the summary file in the backup upload

### PR DESCRIPTION
**Pull request by AbrilRBS**
Changelog: Fix: Don't fail when we can't overwrite the summary file in the backup upload.
Docs: Omit

Missing tests

### CODE REVIEW

File: conan/api/subapi/upload.py [hunk 1]
```diff
@@ -120,17 +120,23 @@ def upload_backup_sources(self, files):
         for file in files:
             basename = os.path.basename(file)
             full_url = url + basename
+            is_summary = file.endswith(".json")
+            file_kind = "summary" if is_summary else "file"
             try:
-                # Always upload summary .json but only upload blob if it does not already exist
-                if file.endswith(".json") or not uploader.exists(full_url, auth=None):
-                    output.info(f"Uploading file '{basename}' to backup sources server")
+                if is_summary or not uploader.exists(full_url, auth=None):
```
Comment (memsharded) [inline conan/api/subapi/upload.py:126]: Do we want to gracefully compare the existing summary file in the remote, and use the comparison for the warning?

**Review by memsharded — APPROVED** [review]
Looks good

**Comment by jave27** [comment]
Thanks, @RubenRBS!

**Review by memsharded — APPROVED** [review]

**Comment by memsharded** [comment]
@jave27 merged, this will be in next Conan 2.5 release

**Pull request merged**